llvm/lib/MC/MCParser/AsmParser.cpp

```diff
@@ -1950,7 +1951,9 @@ bool AsmParser::parseStatement(ParseStatementInfo &Info,
       Lex();
     }
 
-    if (MAI.hasSubsectionsViaSymbols() && CFIStartProcLoc && Sym->isExternal())
+    if (MAI.hasSubsectionsViaSymbols() && CFIStartProcLoc &&
+        Sym->isExternal() &&
+        (!isa<MCSymbolMachO>(Sym) || !cast<MCSymbolMachO>(Sym)->isAltEntry()))
```

Comment:
Since only `MCAsmInfoDarwin` sets `hasSubsectionsViaSymbols`. The `isa<MCSymbolMachO>(Sym)` check is unneeded.